### PR TITLE
Optional flag Cluster.delete_telemetry_data for un-manage cluster

### PR DESCRIPTION
**Pull request by shtripat**
bugzilla: 1582375
tendrl-bug-id: Tendrl/commons#819
Signed-off-by: Shubhendu <shtripat@redhat.com>

### CODE REVIEW

**Comment by shtripat** [comment]
@r0h4n There is no specific BZ for removal of graphite data while un-manage of the cluster. This point came up as part of discussion under https://github.com/Tendrl/commons/issues/819#issuecomment-367886647.

File: app/controllers/clusters_controller.rb [hunk 1]
```diff
@@ -82,10 +82,16 @@ class ClustersController < AuthenticatedUsersController
   post '/clusters/:cluster_id/unmanage' do
     Tendrl.load_node_definitions
     flow = Tendrl::Flow.new('namespace.tendrl', 'UnmanageCluster')
+    body = JSON.parse(request.body.read)
+    body['Cluster.delete_telemetry_data'] = if ['yes', 'no'].include?(body['Cluster.delete_telemetry_data'])
```
Comment (r0h4n) [inline app/controllers/clusters_controller.rb]: The default should be "yes" if no value is supplied

Comment (shtripat) [reply app/controllers/clusters_controller.rb]: Deletion if any data is disruptive action so I thought of keeping default as archival.
@julienlim @nthomas-redhat thoughts?

Comment (julienlim) [reply app/controllers/clusters_controller.rb]: @shtripat We suggested in the design for the default to be to delete the data.  That way, if the user wants to retain/archive the data, it's a conscious decision to do so.

Comment (shtripat) [reply app/controllers/clusters_controller.rb]: @julienlim thanks for confirmation. I will change the logic accordingly.
@r0h4n thanks for pointing this out.

File: app/controllers/clusters_controller.rb [hunk 2]
```diff
@@ -82,10 +82,16 @@ class ClustersController < AuthenticatedUsersController
   post '/clusters/:cluster_id/unmanage' do
     Tendrl.load_node_definitions
     flow = Tendrl::Flow.new('namespace.tendrl', 'UnmanageCluster')
+    body = JSON.parse(request.body.read)
```
Comment (shirshendu) [inline app/controllers/clusters_controller.rb]: This fails if empty JSON object is not provided in the request body.
How about:

```ruby
body = JSON.parse(request.body.read.present? ? request.body.read : '{}')
```

Either this, or else change the failing test to provide an empty JSON while making the call.

Comment (shtripat) [reply app/controllers/clusters_controller.rb]: Ack

**Pull request merged**